reasoner_pydantic/message.py

```diff
@@ -27,16 +28,40 @@ class Message(BaseModel):
         title="knowledge graph",
         nullable=True,
     )
-    results: Optional[HashableSet[Result]] = Field(
+    results: Optional[Results] = Field(
         None,
         title="list of results",
         nullable=True,
     )
+    auxiliary_graphs: Optional[AuxiliaryGraphs] = Field(
+        None,
+        title="list of auxiliary graphs",
+        nullable=True
+    )
 
     class Config:
         title = "message"
         extra = "forbid"
 
+    def from_obj(obj):
```

Comment:
We had a long offline discussion about this. parse_obj somewhat handles this, but it can't handle customs update functions when there are hash collisions. This is the best way we could come up with to ensure all of the merging happens automatically on ingest.

Reply:
Makes sense - maybe worth a comment?

Reply:
Maybe we should include guidance in the README that from_obj should be called in all instances? Otherwise, we could also include it in every model definition. There's no way for us to prevent parse_obj from being called though, unfortunately.

It definitely makes updating to this version a bit more tedious.

Reply:
can we call this parse_obj and then change line 47 to super.parse_obj(obj) ?

Reply:
Alternative solution was found to allow for from_obj to be renamed parse_obj, overwriting parse_obj for the necessary classes.